### PR TITLE
Optimize local builds

### LINKED ISSUE #1873: install-iml-local should cache downloads and build artifacts
`install-iml-local` now builds all the code from scratch every time.
It doesn't cache neither cargo downloads nor the build artifacts.
This is very slow. Where old process cached the `target` directory it was ~3 minutes for complete provision with hot rebuild, now it's 13 minutes for every rebuild.
So we should investigate caching some directories (`$HOME/.cargo` and `iml/target` are 2 specific ones).

### LINKED ISSUE #1880: Optimization of local builds
- [x] don't overwrite `/etc/mock/iml.cfg` every time to not throw off mock's caching of chroot
- [x] overall caching of downloads and artifacts (#1873 )
- [x] use `nosync` (https://github.com/rpm-software-management/mock/wiki/Feature-nosync)
- [x] use yum cache (https://github.com/rpm-software-management/mock/wiki/Plugin-YumCache)
  This is enabled by default

### PR DESCRIPTION
**Pull request by mkpankov**
Bring down `install-iml-local` with hot no-op rebuild to ~4 minutes instead of ~13 minutes.

Fixes #1873 
Fixes #1880

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1881)

### CODE REVIEW

File: vagrant/scripts/install_iml_local.sh [hunk 1]
```diff
@@ -100,13 +111,17 @@ enabled_metadata=1
 """
 EOF
 
+if { ! [ -f $old ]; } || { ! cmp --silent $old $new; } ; then
```
Comment (ip1981) [inline vagrant/scripts/install_iml_local.sh:114]:
```suggestion
if ! cmp --silent $old $new; then
```

File: .copr/Makefile [hunk 2]
```diff
@@ -1,5 +1,6 @@
 BUILDROOT:=$(shell [ -d "/build" ] && echo "/build" || echo ".")
 TMPDIR:=$(shell mktemp -d)
+TARGET:=target
```
Comment (ip1981) [inline .copr/Makefile:3]:
```suggestion
TARGET:=$(or $(CARGO_TARGET_DIR),target)
```

And don't duplicate TARGET in https://github.com/whamcloud/integrated-manager-for-lustre/pull/1881/files#diff-76ac15dc8b7c623d9556dee53bdd8b34R124

File: vagrant/scripts/install_iml_local.sh [hunk 3]
```diff
@@ -1,22 +1,33 @@
 set -x
 
-yum install -y mock-1.2.17
+yum install -y mock-1.2.17 nosync.x86_64
```
Comment (jgrund) [inline vagrant/scripts/install_iml_local.sh:3]:
```suggestion
yum install -y mock-1.2.17 nosync
```

File: vagrant/scripts/install_iml_local.sh (hunk 1 above)
Comment (mkpankov) [reply vagrant/scripts/install_iml_local.sh]: Done

File: .copr/Makefile (hunk 2 above)
Comment (mkpankov) [reply .copr/Makefile]: Done

File: vagrant/scripts/install_iml_local.sh (hunk 3 above)
Comment (mkpankov) [reply vagrant/scripts/install_iml_local.sh]: Done

**Comment by mkpankov** [comment]
> Will it work for 2 different vagrant builds?

What do you mean?

**Comment by jgrund** [comment]
Looks like system tests failed with a network error during repo fetching. Rebuilding as: https://build.whamcloud.com/job/manager-for-lustre/514/

**Pull request merged**